### PR TITLE
kubectl: Support scaling deployments

### PR DESCRIPTION
**Pull request by 0xmichalis**
Fixes https://github.com/kubernetes/kubernetes/issues/17170

@bgrant0607 @ironcladlou @janetkuo @nikhiljindal @soltysh

### CODE REVIEW

**Comment by soltysh** [comment]
LGTM

**Comment by kargakis** [comment]
I've removed the DO NOT MERGE tag. I don't think the 30s window in the controller currently should block this.

File: hack/test-cmd.sh [hunk 1]
```diff
@@ -855,6 +856,15 @@ __EOF__
   # Clean-up
   kubectl delete rc redis-{master,slave} "${kube_flags[@]}"
 
+  ### Scale jobs, deployments
```
Comment (soltysh) [inline hack/test-cmd.sh]: :+1:

Comment (kargakis) [inline hack/test-cmd.sh]: yup, pushed

**Comment by kargakis** [comment]
I've changed `kubectl scale deployments` to use the scaling endpoint for the deployments. PTAL

**Comment by kargakis** [comment]
```
Summarizing 1 Failure:

[Fail] Probing container [It] with readiness probe should not be ready before initial delay and never restart [Conformance] 
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/container_probe.go:57

Ran 113 of 202 Specs in 554.393 seconds
FAIL! -- 112 Passed | 1 Failed | 2 Pending | 87 Skipped 
```
flake(?)

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Comment by nikhiljindal** [comment]
Yes the test seems flaky.
Retrying @k8s-bot ok to test

**Comment by kargakis** [comment]
had to rebase

**Comment by nikhiljindal** [comment]
@kargakis Feel free to readd the LGTM label after rebases.

**Comment by kargakis** [comment]
@nikhiljindal I would if I could:)

**Comment by kargakis** [comment]
one more rebase

**Pull request merged**